corehq/apps/accounting/migrations/0086_add_duplicate_invoice_id_to_invoice_model.py

```diff
@@ -0,0 +1,35 @@
+# Generated by Django 3.2.20 on 2023-09-14 01:13
+
+from django.db import migrations, models
+from django.db.models import Count
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('accounting', '0085_remove_free_50_sms'),
+    ]
+
+    def _populate_duplicate_invoice_id(apps, schema_editor):
+        Invoice = apps.get_model('accounting', 'Invoice')
+        duplicates = (Invoice.objects.filter(is_hidden_to_ops=False)
+            .values('subscription', 'date_start', 'date_end')
+            .annotate(count=Count('id'))
+            .filter(count__gt=1))
+        for duplicate in duplicates:
+            duplicate_invoices = Invoice.objects.filter(subscription=duplicate['subscription'],
+                                                        date_start=duplicate['date_start'],
+                                                        date_end=duplicate['date_end']).order_by('date_created')
+            first_invoice = duplicate_invoices.first()
+            for invoice in duplicate_invoices[1:]:
+                invoice.duplicate_invoice_id = first_invoice.id
+                invoice.save()
```

Comment:
```suggestion
            duplicate_invoices[1:].update(duplicate_invoice_id=first_invoice.id)
```

Reply:
Thank you!! What a coincidence. I just posted a comment to explain and received your review immediately. :joy:
It seems like Django doesn't allow update on sliced queryset.

Reply:
oh, that makes sense. In that case this should work:
```suggestion
            duplicate_invoices.filter(
                ~Q(id=first_invoice.id)
            ).update(duplicate_invoice_id=first_invoice.id)
```
The other suggestion to import `Q` is also needed for this to work.

Reply:
Thank you! It works! Committed!